src/db/ai_help.rs

```diff
@@ -114,6 +114,24 @@ pub fn create_or_increment_limit(
     }
 }
 
+pub fn decrement_limit_total(conn: &mut PgConnection, user: &UserQuery) -> Result<(), DbError> {
+    if user.is_subscriber() {
+        update(limits::table)
+            .filter(limits::user_id.eq(&user.id))
+            .set(limits::total_questions.eq(limits::total_questions - 1))
+            .execute(conn)?;
+    } else {
+        update(limits::table)
+            .filter(limits::user_id.eq(&user.id))
+            .set((
+                (limits::total_questions.eq(limits::total_questions - 1)),
+                (limits::session_questions.eq(limits::session_questions - 1)),
+            ))
+            .execute(conn)?;
+    }
```

Comment:
Afaik we increment `session_question` and `total_questions` for both subscribers and free users, so we should also decrement both. No?

```suggestion
    update(limits::table)
        .filter(limits::user_id.eq(&user.id))
        .set((
            (limits::total_questions.eq(limits::total_questions - 1)),
            (limits::session_questions.eq(limits::session_questions - 1)),
        ))
        .execute(conn)?;
```

Reply:
Sure, I was confused by the original code using `create_or_increment_total` and `create_or_increment_limit` for the different subscriber types, but these are all around actually limiting. Our aim here is not to enforce the limit but set the numbers right.